runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -132,6 +132,14 @@ public final synchronized ILogData nextUpTo(final long maxGlobal) {
         if (entry != null) {
             // Update the pointer.
             updatePointer(entry);
+
+            // If we see a hole that doesn't have a backpointer that belongs to this
+            // stream then it can be skipped. Otherwise, holes that do belong to this
+            // stream (i.e., written by the checkpointer) will be returned and not skipped.
+            // Higher layers can interpret this hole as a no-op
+            if (entry.isHole() && entry.getBackpointerMap().isEmpty()) {
+                return nextUpTo(maxGlobal);
+            }
```

Comment:
It can contain more than one stream if the object has stream tags. For example:

`2022-11-01T07:24:52.354Z | DEBUG |                      LogUnit-3 |            o.c.i.LogUnitServer | handleWrite: HOLE, at: Token(epoch=1, sequence=20327), ids: {65b29143-5d1f-3714-8010-dc7cc2a3e666=-6, 55d9bc21-3684-39b3-94d2-d04fdbbf5515=19302, 2851d785-cfdc-3e87-a51a-33360452e4c8=20256, 846418e4-cf12-313b-9dd6-ba40be3a84ab=19767, 74b85167-cf88-3315-9592-631fac575857=20317}`